lib/wab/impl/handler.rb

```diff
@@ -74,6 +74,18 @@ def extract_req(req)
         [@shell.path_controller(path), path, query, body]
       end
 
+      def compute_result(method, ctrl, path, query, body=nil)
```

Comment:
This approach adds complexity instead of simplifying. You take calls like `ctrl.read` and replace it with a conditional that then calls the exact same function. Make the code more complicated, slower, and harder to follow.